src/main/java/org/jenkinsci/plugins/vSphereCloudProvisionedSlave.java

```diff
@@ -100,89 +100,19 @@ public void preLaunch(Computer c, TaskListener taskListener) throws IOException,
     }
 
     @Extension
-    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
-        @Override
-        public boolean filter(@CheckForNull Object context, @NonNull Descriptor descriptor) {
-            return !(descriptor instanceof DescriptorImpl);
-        }
-    }
-
-    @Extension
-    public static final class DescriptorImpl extends SlaveDescriptor {
-
-        public DescriptorImpl() {
-            load();
-        }
-
+    public static final class DescriptorImpl extends vSphereCloudSlave.DescriptorImpl {
         @Override
         public String getDisplayName() {
-            return "Slave created from a vSphere Cloud slave template";
+            return super.getDisplayName()+", auto-provisioned by Jenkins from cloud template";
         }
 
         @Override
         public boolean isInstantiable() {
+            /*
+             * This type of agent can't be directly created by the user through the UI.
+             * The user defines a vSphere agent template and _that_ then creates these "on demand".
+             */
             return false;
```

Comment:
I think we are fine as long as this is kept in. This is meant to prevent JCasC from removing nodes provisioned from this class.